sotodlib/coords/helpers.py

```diff
@@ -228,17 +244,45 @@ def get_wcs_kernel(proj, ra, dec, res):
     Returns a WCS object that captures the requested pixelization.
 
     """
-    assert np.isscalar(res)  # This ain't enlib.
-    _, wcs = enmap.geometry(np.array((dec, ra)), shape=(1,1), proj=proj,
-                            res=(res, -res))
+    if len(proj) > 4:
+        m = re.match('(?P<proj>car|tan|cea|gnom)-(?P<res>[0-9]*.?[0-9]*)(?P<unit>arcsec|arcmin|deg)',
+                     proj)
+        if m is None:
+            raise ValueError("Input projection string is incorrectly formatted.")
+        proj = m['proj']
+        res = float(m['res'])
+        unit = m['unit']
+
+        # Convert to radians
+        res = res * np.pi/180.
+        if unit == 'arcmin':
+            res /= 60.
+        if unit == 'arcsec':
+            res /= 3600.
+
+        _, wcs = enmap.geometry(np.array((0, 0)), shape=(1, 1), proj=proj,
+                                res=(res, -res))
+
+    else:
+        assert np.isscalar(res)  # This ain't enlib.
+        _, wcs = enmap.geometry(np.array((dec, ra)), shape=(1, 1), proj=proj,
+                                res=(res, -res))
+
+    if proj == 'car':
+        wcs.wcs.crpix = [1.0, 0.5]
+
     return wcs
 
+
 def get_footprint(tod, wcs_kernel, dets=None, timestamps=None, boresight=None,
                   focal_plane=None, sight=None, rot=None):
     """Find a geometry (in the sense of enmap) based on wcs_kernel that is
     big enough to contain all data from tod.  Returns (shape, wcs).
 
     """
+    if type(wcs_kernel) == str:
+        wcs_kernel = get_wcs_kernel(wcs_kernel)
+
```

Comment:
Python people try not to be too strict about classes.  We don't need this to be a str, we just need it to be sufficiently str-like.  I.e. any subclass of a str is acceptable.  So ... replace the check with `if isinstance(wcs_kernel, str)`.

This is not likely to ruin anyone's day in this case, but generally it's a good practice when all you're trying to do is distinguish between broad classes.

Reply:
Makes sense. Done